dev/release/06-java-upload.sh

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env bash
```

Comment:
How about making https://github.com/apache/arrow/blob/master/dev/release/06-java-upload.sh reusable from other repositories like https://github.com/apache/arrow/blob/master/dev/release/05-binary-upload.sh instead of coping this file?

I can work on it.

Reply:
Sounds good to me. That would be much appreciated.

Reply:
OK. I work on it: https://github.com/apache/arrow/issues/14816

Reply:
https://github.com/apache/arrow-adbc/pull/213